trading_calendars/exchange_calendar_xphs.py

```diff
@@ -0,0 +1,425 @@
+#
+# Copyright 2019 Quantopian, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from datetime import time, timedelta
+from itertools import chain
+import pandas as pd
+from pandas.tseries.holiday import (
+    Holiday,
+    GoodFriday,
+)
+from pytz import timezone
+
+from .common_holidays import (
+    european_labour_day,
+    new_years_day,
+    maundy_thursday,
+    christmas,
+    christmas_eve,
+    new_years_eve,
+    all_saints_day,
+    chinese_lunar_new_year_dates,
+)
+from .trading_calendar import (
+    TradingCalendar,
+    HolidayCalendar,
+    MONDAY,
+    FRIDAY,
+)
+
+
+def only_friday(dt):
+    """
+    Only keeps the holidays that fall on a Friday.  Useful
+    for defining holidays that add a Friday to make a 4-day weekend
+    when falling on a Thursday.
+    """
+    return dt[dt.weekday == FRIDAY]
+
+
+def previous_monday(dt):
+    """
+    Returns previous monday on or before the given date.
+    """
+    while dt.weekday() != MONDAY:
+        dt -= timedelta(days=1)
+    return dt
+
+
+# All pre-2011 holidays are pre-computed, so we define Holidays starting
+# in 2011.
+NewYearsDay = new_years_day(start_date='2011')
+
+# Chinese New Year's only observed after 2011
+ChineseNewYear = chinese_lunar_new_year_dates
+
+ChineseNewYearAfter2011 = ChineseNewYear[ChineseNewYear.year > 2011]
```

Comment:
Did you mean `>= 2011`? Wondering because all of the other holidays start in 2011, but this is starting in 2012.

Reply:
I wondered about this too, but they didn't celebrate Chinese New Year in 2011 either.